Adds rubber ducky shoes

kugamo opened the pull request:
<!-- Write **BELOW** The Headers and **ABOVE** The comments else it may not be viewable. -->
<!-- You can view Contributing.MD for a detailed description of the pull request process. -->

## What Does This PR Do
This PR adds rubber ducky shoes as an item. It can be worn on the feet, and makes squeaks everytime you walk. It can be crafted using a wirecutters with 2 rubber duckies, and a pair of sandals.
<!-- Include a small to medium description of what your PR changes. Document every changes or this may delay review or even discourage maintainers from merging your PR! -->

## Why It's Good For The Game
Its stupid fun, plain and simple. Ideally this would add to the charm of the game.
<!-- Please add a short description of why you think these changes would benefit the game. If you can't justify it in words, it might not be worth adding. -->

## Images of changes
![PaintDotNet_sv3YZ07ZJi](https://user-images.githubusercontent.com/16618648/89138302-cdf01100-d500-11ea-8ca2-3864e9a69820.png)
![ducker](https://user-images.githubusercontent.com/16618648/89860955-5195a780-db6a-11ea-845c-bf8aaeeb597e.png)
<!-- If you did not make a map or sprite edit, you may delete this section. You may include a gif of your feature if you want -->

## Changelog
:cl:
add: Adds Rubber Ducky Shoes, and a crafting recipe for them.
tweak: Adds Rubber Duckies as a prize in the arcade for 80.
imageadd: Adds sprites for item icon, on character, and in hand.
/:cl:

<!-- Both :cl:'s are required for the changelog to work! You can put your name to the right of the first :cl: if you want to overwrite your GitHub username as author ingame. -->
<!-- You can use multiple of the same prefix (they're only used for the icon ingame) and delete the unneeded ones. Despite some of the tags, changelogs should generally represent how a player might be affected by the changes rather than a summary of the PR's contents. -->

Mitchs98 commented:
Should definitely add more ways to get rubber duckies if these get merged. Maybe not an incredibly easy way...but it'd be pretty hard to get a pair of these otherwise.

SteelSlayer commented on code/modules/clothing/shoes/miscellaneous.dm:
```diff
@@ -343,3 +343,11 @@
+	desc = "Two rubber duckies made to fit."
+	icon_state = "ducky"
+	item_state = "ducky"
+	var/footstep = 1
```
What is this var for? It's not used anywhere.

kugamo replied:
well I know next to nothing about byond code, I just noticed that for the other shoes with sounds, like jackboots and clownshoes, had it. So I copied it over. I never tested to see if it still made the noises without it, I just assumed it was necessary.

kugamo replied:
OK so sounds work fine without that var, and it is seemingly totally pointless. (for rubber ducky shoes at least)

SteelSlayer replied:
Aha I see. After some research, those vars on the other shoes are also unused, because shoe step sound is handled elsewhere and it never even uses the `footstep` var.

kugamo replied:
must be an artifact or something

kugamo commented:
![PaintDotNet_sg2u5vYuSX](https://user-images.githubusercontent.com/16618648/89148184-b9bb0c80-d51e-11ea-8055-86fa9db30d85.png)
![dreamseeker_7sMR6oiIk9](https://user-images.githubusercontent.com/16618648/89148193-bfb0ed80-d51e-11ea-89d7-8599651532f8.png)
I think this covers all issues for now, I am treating this as finished for now. Unless I need to change the merch console additons, its done.

taukausanake commented:
A soap or wash crate wouldn't be bad. Maybe include with the janitor crate. Just a rubber ducky crate wouldn't be good unless it came with more than a couple but too much wouldn't be good either

datlo commented:
I think it'd be better to have them be in a cargo crate, that way there's a way to regulate/slow down ducky spam.

farie82 commented on code/modules/clothing/shoes/miscellaneous.dm:
```diff
@@ -343,3 +343,10 @@
+	desc = "These shoes are made for quacking, and thats just what they'll do."
+	icon_state = "ducky"
+	item_state = "ducky"
+	shoe_sound = list('sound/items/squeaktoy.ogg' = 1)
```
`shoe_sound` does not expect a list. It expects a string which is the sound.
Example
![image](https://user-images.githubusercontent.com/15887760/89191018-9a65c300-d5a2-11ea-95ed-2c80d983174a.png)

kugamo replied:
I had trouble getting sounds using shoe_sound = squeaktoy or shoe_sound = squeaktoy.ogg, so I tried using what code I found for the rubberducky (which used a list for whatever reason). After some testing I found that using a filepath works fine, so I just commmited that.

Spacemanspark commented:
I can't deny this earned a chuckle from me.
I do think it shouldn't be _too_ easy to get so there's not a billion spamming ducklings running around, but I like this.
:+1:

kugamo commented:
So I removed rubber ducky as an entry in the merch console, and added it to the arcade. I moved it to the arcade because ~~I couldn't find where the crate code is~~ its more of a hurtle to get it than just $300, and I still want it some what accessible. You would still have to get 2 to make the shoes, which in total amounts to 160 tickets.
![dreamseeker_XjRLpNmhxb](https://user-images.githubusercontent.com/16618648/89238196-de64c200-d5ba-11ea-9588-a6b41e4c2270.png)

KasparoVy commented:
The quackers, they're adorable

variableundefined reviewed (approved):
@kugamo Will merge when deconflicted.

Spent 2 minutes running around in circle in an empty station with this stuff.
